pkg/virt-operator/resource/apply/crds.go

```diff
@@ -47,42 +48,47 @@ func (r *Reconciler) createOrUpdateCrd(crd *extv1beta1.CustomResourceDefinition)
 		return nil
 	}
 
-	if !objectMatchesVersion(&cachedCrd.ObjectMeta, version, imageRegistry, id, r.kv.GetGeneration()) {
-		// Patch if old version
-		var ops []string
+	modified := resourcemerge.BoolPtr(false)
+	existing := cachedCrd.DeepCopy()
 
-		// Add Labels and Annotations Patches
-		labelAnnotationPatch, err := createLabelsAndAnnotationsPatch(&crd.ObjectMeta)
-		if err != nil {
-			return err
-		}
-		ops = append(ops, labelAnnotationPatch...)
+	resourcemerge.EnsureCustomResourceDefinitionV1Beta1(modified, existing, *crd)
+	// there was no change to metadata, the generation was right
```

Comment:
Maybe this comment is a copy and paste from somewhere else? nothing it's checking the generation, and the function is checking the whole CRD.